hermes-common/src/main/java/pl/allegro/tech/hermes/common/message/wrapper/AvroMessageContentWrapper.java

```diff
@@ -64,8 +65,11 @@ private MessageMetadata getMetadata(Map<Utf8, Utf8> metadata) {
             GenericRecord genericRecord = bytesToRecord(message, schema);
             genericRecord.put(METADATA_MARKER, metadataMap(id, timestamp, externalMetadata));
             return recordToBytes(genericRecord, schema);
-        } catch (Exception exception) {
-            throw new WrappingException("Could not wrap avro message", exception);
+        } catch (Exception e) {
+            if (e instanceof AvroRuntimeException && e.getMessage().equals("Not a valid schema field: __metadata")) {
+                throw new AvroInvalidMetadataException("Avro schema has invalid field __metadata", e);
+            }
+            throw new WrappingException("Could not wrap avro message", e);
```

Comment:
can't we just stick with `WrappingException` and extend it's message with the cause message?
Like: `throw new WrappingException(String.format("Could not wrap avro message: %s", e.getMessage()), e);`
I think the original message that goes with AvroRuntimeException is just fine, plus we'll avoid equals checking, etc. of the Avro's internal exception.

Reply:
I know it would look better, but I checked usages of `AvroRuntimeException` in the `org.apache.avro:avro:1.8.1` and this library throws this exception 109 times. 
In most cases the exception is thrown due to client error, but this exception is also used to wrap caught `Exception`s and `IOException`s, that's why I decided to handle this one specific case.